examples/spec/query_spec.rb

```diff
@@ -0,0 +1,84 @@
+require_relative 'spec_helper'
+require_relative '../lib/query'
+
+RSpec.describe Query do
+  include RSpec::Hive::WithHiveConnection
+  include RSpec::Hive::QueryBuilderHelper
+
+  subject { described_class.new }
+
+  let(:schema) { subject.table_schema }
+
+  before { connection.execute(schema.create_table_statement) }
+
+  context 'without stubbing strategy' do
+    let(:input_data) do
+      [
+        ['Mikolaj', 'Cos', 1.23, 1],
+        ['Wojtek', 'Cos', 3.76, 2]
+      ]
+    end
+    let(:query_result) { connection.fetch(query) }
+
+    before { into_hive(schema).insert(*input_data).execute }
+
+    context 'when querying for amount > 3.2' do
+      let(:query) { "SELECT * FROM `#{subject.table_name}` WHERE amount > 3.2" }
+      let(:expected_result_array) do
+        [
+          [a_string_matching('Wojtek'), 'Cos', be_within(0.01).of(3.76)]
+        ]
+      end
+
+      it 'returns Wojtek' do
+        expect(query_result).to match_result_set(expected_result_array)
+      end
+    end
+
+    context 'when querying for amount < 3.2' do
+      let(:query) { "SELECT * FROM `#{subject.table_name}` WHERE amount < 3.2" }
+      let(:expected_result_hash) do
+        [
+          {name: 'Mikolaj', address: 'Cos', amount: be_within(0.01).of(1.23) }
+        ]
+      end
+
+      it 'returns Mikolaj' do
+        expect(query_result).to match_result_set(expected_result_hash)
+      end
+    end
+  end
+
+  context 'with stubbing strategy' do
```

Comment:
It's not clear to me what's the difference between 'with stubbing" and 'without stubbing'. Could you elaborate?